### ISSUE TITLE
Cannot install 'vagrant-puppet-install' on Windows 7

### ISSUE THREAD

**Issue by jeff1evesque** [petems/vagrant-puppet-install#25]
I am running windows 7 host (Ubuntu 14.04 guest).  I am unable to implement my [`Vagrantfile`](https://github.com/jeff1evesque/machine-learning/blob/474ed3580bd821db49bfc51806cec553585108c7/Vagrantfile):

```
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure(2) do |config|
  # The most common configuration options are documented and commented below.
  # For a complete reference, please see the online documentation at
  # https://docs.vagrantup.com.

  ## Variables (ruby syntax)
  required_plugins = %w(vagrant-r10k vagrant-puppet-install)
  plugin_installed = false

  ## Install Vagrant Plugins
  required_plugins.each do |plugin|
    unless Vagrant.has_plugin? plugin
      system "vagrant plugin install #{plugin}"
      plugin_installed = true
    end
  end

  ## Restart Vagrant: if new plugin installed
  if plugin_installed == true
    exec "vagrant #{ARGV.join(' ')}"
  end

  ## Every Vagrant development environment requires a box. You can search for
  #  boxes at https://atlas.hashicorp.com/search.
  config.vm.box = "ubuntu/trusty64"
  
  ## Update puppet
  config.puppet_install.puppet_version = '4.1.0'

  ## Run r10k
  config.r10k.puppet_dir = 'puppet'
  config.r10k.puppetfile_path = 'puppet/Puppetfile'
  
  ## Custom Manifest: install needed packages
  #
  #  Note: future parser allow array iteration in the puppet manifest
  config.vm.provision "puppet" do |puppet|
    puppet.manifests_path = "puppet/manifests"
    puppet.manifest_file  = "install_packages.pp"
    puppet.module_path    = "puppet/modules"
    puppet.options        = ["--parser", "future"]
  end

  ## Custom Manifest: build scikit-learn
  config.vm.provision "puppet" do |puppet|
    puppet.manifests_path = "puppet/manifests"
    puppet.manifest_file  = "install_sklearn.pp"
    puppet.module_path    = "puppet/modules"
  end

  ## Custom Manifest: install, and configure SQL database
  config.vm.provision "puppet" do |puppet|
    puppet.manifests_path = "puppet/manifests"
    puppet.manifest_file  = "setup_database.pp"
    puppet.module_path    = "puppet/modules"
  end
  
  config.vm.network "forwarded_port", guest: 5000, host: 8080
  config.vm.network "forwarded_port", guest: 443, host: 8585
end
```

I get the following error traceback:

```bash
jeffrey@my-pc ~/web-projects/machine-learning (vagrant_puppet_install)
$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: vagrant-r10k: Building the r10k module path with puppet provisioner
 module_path "puppet/modules". (if module_path is an array, first element is use
d)
==> default: vagrant-r10k: Beginning r10k deploy of puppet modules into c:/Users
/Jeff/web-projects/machine-learning/puppet/modules using c:/Users/Jeff/web-projects/machine-learning/puppet/Puppetfile
==> default: vagrant-r10k: Deploy finished
==> default: Importing base box 'ubuntu/trusty64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/trusty64' is up to date...
==> default: Setting the name of the VM: machine-learning_default_1436399519686_40523
==> default: Clearing any previously set forwarded ports...
==> default: vagrant-r10k: Building the r10k module path with puppet provisioner
 module_path "puppet/modules". (if module_path is an array, first element is used)
==> default: vagrant-r10k: Beginning r10k deploy of puppet modules into c:/Users
/Jeff/web-projects/machine-learning/puppet/modules using c:/Users/Jeff/web-projects/machine-learning/puppet/Puppetfile
==> default: vagrant-r10k: Deploy finished
==> default: Fixed port collision for 22 => 2222. Now on port 2200.
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 5000 => 8080 (adapter 1)
    default: 443 => 8585 (adapter 1)
    default: 22 => 2200 (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2200
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if its present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
==> default: Mounting shared folders...
    default: /vagrant => C:/Users/Jeff/web-projects/machine-learning
    default: /tmp/vagrant-puppet/modules-6a27f8975c46c0faa72bb7408a168135 => C:/Users/Jeff/web-projects/machine-learning/puppet/modules
    default: /tmp/vagrant-puppet/manifests-846018e2aa141a5eb79a64b4015fc5f3 => C:/Users/Jeff/web-projects/machine-learning/puppet/manifests
==> default: Installing Puppet 4.1.0 package...
==> default: 23:53:00 +0000 INFO: Version parameter defined: 4.1.0
==> default: 23:53:00 +0000 INFO: Downloading Puppet 4.1.0 for ubuntu...
==> default: 23:53:00 +0000 INFO: Ubuntu platform! Lets get you a DEB...
==> default: 23:53:00 +0000 INFO: Downloading http://apt.puppetlabs.com/puppetla
bs-release-trusty.deb
==> default: 23:53:00 +0000 INFO:   to file /tmp/install.sh.1823.21267/puppetlabs-release-trusty.deb
==> default: 23:53:00 +0000 INFO: Trying wget...
==> default: 23:53:00 +0000 INFO: installing with dpkg...
==> default: Selecting previously unselected package puppetlabs-release.
==> default: (Reading database ... 86712 files and directories currently installed.)
==> default: Preparing to unpack .../puppetlabs-release-trusty.deb ...
==> default: Unpacking puppetlabs-release (1.0-11) ...
==> default: Setting up puppetlabs-release (1.0-11) ...
==> default: Ign http://apt.puppetlabs.com trusty InRelease
==> default: Ign http://security.ubuntu.com trusty-security InRelease
==> default: Get:1 http://apt.puppetlabs.com trusty Release.gpg [876 B]
==> default: Ign http://archive.ubuntu.com trusty InRelease
==> default: Get:2 http://security.ubuntu.com trusty-security Release.gpg [933 B]
==> default: Get:3 http://apt.puppetlabs.com trusty Release [29.3 kB]
==> default: Ign http://archive.ubuntu.com trusty-updates InRelease
==> default: Get:4 http://security.ubuntu.com trusty-security Release [63.5 kB]
==> default: Hit http://archive.ubuntu.com trusty Release.gpg
==> default: Get:5 http://archive.ubuntu.com trusty-updates Release.gpg [933 B]
==> default: Get:6 http://apt.puppetlabs.com trusty/main Sources [21.4 kB]
==> default: Hit http://archive.ubuntu.com trusty Release
==> default: Get:7 http://apt.puppetlabs.com trusty/dependencies Sources [1,418B]
==> default: Get:8 http://archive.ubuntu.com trusty-updates Release [63.5 kB]
==> default: Get:9 http://security.ubuntu.com trusty-security/main Sources [87.9 kB]
==> default: Get:10 http://apt.puppetlabs.com trusty/main amd64 Packages [32.2 kB]
==> default: Get:11 http://apt.puppetlabs.com trusty/dependencies amd64 Packages [932 B]
==> default: Get:12 http://security.ubuntu.com trusty-security/universe Sources[28.1 kB]
==> default: Get:13 http://archive.ubuntu.com trusty/main Sources [1,064 kB]
==> default: Get:14 http://security.ubuntu.com trusty-security/main amd64 Packages [310 kB]
==> default: Get:15 http://security.ubuntu.com trusty-security/universe amd64 Packages [111 kB]
==> default: Get:16 http://security.ubuntu.com trusty-security/main Translation-en [164 kB]
==> default: Get:17 http://security.ubuntu.com trusty-security/universe Translation-en [64.0 kB]
==> default: Ign http://apt.puppetlabs.com trusty/dependencies Translation-en_US

==> default: Ign http://apt.puppetlabs.com trusty/dependencies Translation-en
==> default: Get:18 http://archive.ubuntu.com trusty/universe Sources [6,399 kB]

==> default: Ign http://apt.puppetlabs.com trusty/main Translation-en_US
==> default: Ign http://apt.puppetlabs.com trusty/main Translation-en
==> default: Hit http://archive.ubuntu.com trusty/main amd64 Packages
==> default: Hit http://archive.ubuntu.com trusty/universe amd64 Packages
==> default: Hit http://archive.ubuntu.com trusty/main Translation-en
==> default: Hit http://archive.ubuntu.com trusty/universe Translation-en
==> default: Get:19 http://archive.ubuntu.com trusty-updates/main Sources [213 kB]
==> default: Get:20 http://archive.ubuntu.com trusty-updates/universe Sources [125 kB]
==> default: Get:21 http://archive.ubuntu.com trusty-updates/main amd64 Packages [568 kB]
==> default: Get:22 http://archive.ubuntu.com trusty-updates/universe amd64 Packages [295 kB]
==> default: Get:23 http://archive.ubuntu.com trusty-updates/main Translation-en [274 kB]
==> default: Get:24 http://archive.ubuntu.com trusty-updates/universe Translation-en [156 kB]
==> default: Ign http://archive.ubuntu.com trusty/main Translation-en_US
==> default: Ign http://archive.ubuntu.com trusty/universe Translation-en_US
==> default: Fetched 10.1 MB in 9s (1,118 kB/s)
==> default: Reading package lists...
←[1m←[0m
==> default: Reading package lists...
←[1m←[0m
==> default: Building dependency tree...
←[1m←[0m
==> default: Reading state information...
←[1m←[0m
==> default: E
==> default: :
==> default: Version '4.1.0-1puppetlabs1' for 'puppet-common' was not found
←[1m←[0m
==> default: E
==> default: :
==> default: Version '4.1.0-1puppetlabs1' for 'puppet' was not found
←[1m←[0m
==> default: 23:53:11 +0000 CRIT: Installation failed
==> default: 23:53:11 +0000 CRIT: Please file a bug report at https://github.com
/petems/puppet-install-shell/
==> default: 23:53:11 +0000 CRIT:
==> default: 23:53:11 +0000 CRIT: Version: 4.1.0
==> default: 23:53:11 +0000 CRIT: Platform: ubuntu
==> default: 23:53:11 +0000 CRIT: Platform Version: 14.04
==> default: 23:53:11 +0000 CRIT: Machine: x86_64
==> default: 23:53:11 +0000 CRIT: OS: Linux
==> default: 23:53:11 +0000 CRIT:
==> default: 23:53:11 +0000 CRIT: Please detail your operating system type, vers
ion and any other relevant details
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

sh install.sh -v 4.1.0 2>&1

Stdout from the command:

23:53:00 +0000 INFO: Version parameter defined: 4.1.0
23:53:00 +0000 INFO: Downloading Puppet 4.1.0 for ubuntu...
23:53:00 +0000 INFO: Ubuntu platform! Lets get you a DEB...
23:53:00 +0000 INFO: Downloading http://apt.puppetlabs.com/puppetlabs-release-tr
usty.deb
23:53:00 +0000 INFO:   to file /tmp/install.sh.1823.21267/puppetlabs-release-tru
sty.deb
23:53:00 +0000 INFO: Trying wget...
23:53:00 +0000 INFO: installing with dpkg...
Selecting previously unselected package puppetlabs-release.
(Reading database ... 86712 files and directories currently installed.)
Preparing to unpack .../puppetlabs-release-trusty.deb ...
Unpacking puppetlabs-release (1.0-11) ...
Setting up puppetlabs-release (1.0-11) ...
Ign http://apt.puppetlabs.com trusty InRelease
Ign http://security.ubuntu.com trusty-security InRelease
Get:1 http://apt.puppetlabs.com trusty Release.gpg [876 B]
Ign http://archive.ubuntu.com trusty InRelease
Get:2 http://security.ubuntu.com trusty-security Release.gpg [933 B]
Get:3 http://apt.puppetlabs.com trusty Release [29.3 kB]
Ign http://archive.ubuntu.com trusty-updates InRelease
Get:4 http://security.ubuntu.com trusty-security Release [63.5 kB]
Hit http://archive.ubuntu.com trusty Release.gpg
Get:5 http://archive.ubuntu.com trusty-updates Release.gpg [933 B]
Get:6 http://apt.puppetlabs.com trusty/main Sources [21.4 kB]
Hit http://archive.ubuntu.com trusty Release
Get:7 http://apt.puppetlabs.com trusty/dependencies Sources [1,418 B]
Get:8 http://archive.ubuntu.com trusty-updates Release [63.5 kB]
Get:9 http://security.ubuntu.com trusty-security/main Sources [87.9 kB]
Get:10 http://apt.puppetlabs.com trusty/main amd64 Packages [32.2 kB]
Get:11 http://apt.puppetlabs.com trusty/dependencies amd64 Packages [932 B]
Get:12 http://security.ubuntu.com trusty-security/universe Sources [28.1 kB]
Get:13 http://archive.ubuntu.com trusty/main Sources [1,064 kB]
Get:14 http://security.ubuntu.com trusty-security/main amd64 Packages [310 kB]
Get:15 http://security.ubuntu.com trusty-security/universe amd64 Packages [111 kB]
Get:16 http://security.ubuntu.com trusty-security/main Translation-en [164 kB]
Get:17 http://security.ubuntu.com trusty-security/universe Translation-en [64.0kB]
Ign http://apt.puppetlabs.com trusty/dependencies Translation-en_US
Ign http://apt.puppetlabs.com trusty/dependencies Translation-en
Get:18 http://archive.ubuntu.com trusty/universe Sources [6,399 kB]
Ign http://apt.puppetlabs.com trusty/main Translation-en_US
Ign http://apt.puppetlabs.com trusty/main Translation-en
Hit http://archive.ubuntu.com trusty/main amd64 Packages
Hit http://archive.ubuntu.com trusty/universe amd64 Packages
Hit http://archive.ubuntu.com trusty/main Translation-en
Hit http://archive.ubuntu.com trusty/universe Translation-en
Get:19 http://archive.ubuntu.com trusty-updates/main Sources [213 kB]
Get:20 http://archive.ubuntu.com trusty-updates/universe Sources [125 kB]
Get:21 http://archive.ubuntu.com trusty-updates/main amd64 Packages [568 kB]
Get:22 http://archive.ubuntu.com trusty-updates/universe amd64 Packages [295 kB]

Get:23 http://archive.ubuntu.com trusty-updates/main Translation-en [274 kB]
Get:24 http://archive.ubuntu.com trusty-updates/universe Translation-en [156 kB]

Ign http://archive.ubuntu.com trusty/main Translation-en_US
Ign http://archive.ubuntu.com trusty/universe Translation-en_US
Fetched 10.1 MB in 9s (1,118 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
E: Version '4.1.0-1puppetlabs1' for 'puppet-common' was not found
E: Version '4.1.0-1puppetlabs1' for 'puppet' was not found
23:53:11 +0000 CRIT: Installation failed
23:53:11 +0000 CRIT: Please file a bug report at https://github.com/petems/puppe
t-install-shell/
23:53:11 +0000 CRIT:
23:53:11 +0000 CRIT: Version: 4.1.0
23:53:11 +0000 CRIT: Platform: ubuntu
23:53:11 +0000 CRIT: Platform Version: 14.04
23:53:11 +0000 CRIT: Machine: x86_64
23:53:11 +0000 CRIT: OS: Linux
23:53:11 +0000 CRIT:
23:53:11 +0000 CRIT: Please detail your operating system type, version and any other relevant details


Stderr from the command:

stdin: is not a tty
```

While this plugin does not work, I will implement [`puppet_updater.sh`](https://github.com/jeff1evesque/machine-learning/blob/e35979b6eb941b59a5d70300644b46569c8d45d8/puppet/scripts/puppet_updater.sh), which updates puppet to [3.8](https://puppetlabs.com/blog/introducing-puppet-enterprise-3.8):

```bash
#!/bin/bash

if [ ! -e /tmp/puppet-updated ]; then
  wget -O /tmp/puppetlabs-release-trusty64.deb http://apt.puppetlabs.com/puppetlabs-release-trusty.deb
  dpkg -i /tmp/puppetlabs-release-trusty64.deb
  apt-get update
  apt-get --assume-yes install puppet
  touch /tmp/puppet-updated
fi
```

and call it from my [`Vagrantfile`](https://github.com/jeff1evesque/machine-learning/blob/e35979b6eb941b59a5d70300644b46569c8d45d8/Vagrantfile#L34-L35):

```
...
  ## Update latest version of puppet
  config.vm.provision :shell, :path => "puppet/bash/puppet_updater.sh"
...
```

**Comment by jamiejackson** [comment]
Same problem--Windows 7 with an Ubuntu *12.04* guest:
```
18:35:30 +0000 CRIT: Installation failed
18:35:30 +0000 CRIT: Please file a bug report at https://github.com/petems/puppet-install-shell/
18:35:30 +0000 CRIT:
18:35:30 +0000 CRIT: Version: 4.2.2
18:35:30 +0000 CRIT: Platform: ubuntu
18:35:30 +0000 CRIT: Platform Version: 12.04
18:35:30 +0000 CRIT: Machine: x86_64
18:35:30 +0000 CRIT: OS: Linux
18:35:30 +0000 CRIT:
18:35:30 +0000 CRIT: Please detail your operating system type, version and any other relevant details
```

**Comment by petems** [comment]
Right now the plugin does not work for Puppet > 4.0. See: 

https://github.com/petems/vagrant-puppet-install/issues/29

Will be fixed soon! :smile:

**Comment by jeff1evesque** [comment]
Will the update include a cross linux distro fix, as well as windows 7+?  Wouldn't mind removing my [`puppet_updater.sh`](https://github.com/jeff1evesque/machine-learning/blob/e35979b6eb941b59a5d70300644b46569c8d45d8/puppet/scripts/puppet_updater.sh) implementation.

**Comment by petems** [comment]
@jeff1evesque Yep, the update will work for all supported distros for 4.0, https://docs.puppetlabs.com/guides/platforms.html

**Comment by petems** [comment]
Fixed in #30
